draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -1555,29 +1564,35 @@ user agent might wish to block sending cookies during "third-party" requests
 from setting cookies (see {{third-party-cookies}}).
 
 If the user agent does attach a Cookie header field to an HTTP request, the
-user agent MUST send the cookie-string as the value of the
-header field. The cookie-string is computed as defined in {{retrieval-algorithm}},
-whereby the retrieval-uri is defined as the request-uri and the same-site status
-is computed for the HTTP request as defined in {{same-site-requests}}.
+user agent MUST compute the cookie-string following the algorithm defined in
+{{retrieval-algorithm}}, indicating that the retrieval is for an HTTP request.
+The retrieval-uri is defined as the request-uri and the same-site status of the
+retrieval is computed for the HTTP request as defined in
+{{same-site-requests}}.
 
 ### Non-HTTP APIs {#non-http}
 
 The user agent MAY implement "non-HTTP" APIs that can be used to access
 stored cookies.
 
-A user agent MAY return an empty cookie-string in certain contexts, such as when
-retrieval occurs within a third-party context (see {{third-party-cookies}}).
+A user agent MAY return an empty cookie-string in certain contexts, such as
+when a retrieval occurs within a third-party context (see
+{{third-party-cookies}}).
 
-If a user agent does return cookies for a given call to a "non-HTTP" API with an
-associated Document, then the user agent MUST compute the cookie-string following
-the algorithm defined in {{retrieval-algorithm}, whereby the retrieval-uri is the
-associated Document's URL. The retrieval is same-site if the Document's "site for
-cookies" is same-site with the top-level origin as defined in {{document-requests}}.
+If a user agent does return cookies for a given call to a "non-HTTP" API with
+an associated Document, then the user agent MUST compute the cookie-string
+following the algorithm defined in {{retrieval-algorithm}}, indicating that the
+retrieval is from a "non-HTTP" API. The retrieval-uri is the associated
+Document's cookie URL {{COOKIE-URL}}, and the retrieval is same-site if the
+Document's "site for cookies" is same-site with the top-level origin as defined
+in {{document-requests}}.
 
 ### Retrieval Algorithm {#retrieval-algorithm}
 
-Given a cookie store, a retrieval-uri, and the same-site status of the
-retrieval-uri, the following algorithm returns a cookie-string.
+The following algorithm returns a cookie-string from a given cookie store.
+Callers of this algorithm MUST provide a retrieval-uri, indicate the same-site
+status of the retrieval, and indicate whether the retrieval is for an HTTP
+request or from a "non-HTTP" API.
```

Comment:
nit: Not a huge deal, but I'm not sure the "MUST" thing applies here? At least it reads a bit awkwardly to me. I think the MUST keyword applies when something has to be explicitly marked as required that the UA or server has to do. This algorithm just doesn't make sense if those arguments are not provided, so it's implicitly a requirement to provide them.

I think this would be cleaner if the other things were properties of a retrieval, e.g. retrieval-uri, same-site status, HTTP-flag (I think it might be worth defining a term like HTTP-flag to refer to the HTTP request vs non-HTTP API bit, to refer to it more easily). (So you could say something like "Given a cookie store and a retrieval, the following algorithm ...".)